packages/wrangler/src/index.tsx

```diff
@@ -195,58 +195,94 @@ export async function main(argv: string[]): Promise<void> {
       const destination = path.join(process.cwd(), "wrangler.toml");
       if (fs.existsSync(destination)) {
         console.warn(`${destination} file already exists!`);
-        const result = await confirm(
+        const shouldContinue = await confirm(
           "Do you want to continue initializing this project?"
         );
-        if (!result) {
+        if (!shouldContinue) {
           return;
         }
+      } else {
```

Comment:
Perhaps, if there is a `wrangler.toml` already but the user says to continue, we should load it and check that it does indeed contain the required `compatibilityDate`?

Also, now that you can specify the `compatibilityDate` on the command line, we might not actually need a `wrangler.toml` file at all any more 🤔

Reply:
mentioned this above - I don't think `wrangler init` should validate existing `wrangler.toml` files, that should bet he purview of commands that consume the config. 
I think leaving the field in the comnfig is fine, mentally I think of it like the "engines" field in package.json. Open to revising that of course.

Reply:
I think that one should expect that after running `wrangler init` the project should be setup to work correctly?

Happy to discuss and iterate on this in a later PR.

Reply:
I do notice now that `npm init` does simply overwrite existing `package.json`. Which... isn't great behaviour imo, but it does set precedent. Let's discuss.